tests/jballerina-unit-test/src/test/resources/test-src/reachability-analysis/reachability_analysis2.bal

```diff
@@ -0,0 +1,1034 @@
+// Copyright (c) 2022 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+function executeTestReachabilityWithIfContainingReturn() {
+    assertEqual(testReachabilityWithIfContainingReturn(), "Ballerina");
+    assertEqual(testReachabilityWithIfContainingReturn2(), "Ballerina");
+    assertEqual(testReachabilityWithIfContainingReturn3(), "Hello");
+    assertEqual(testReachabilityWithIfContainingReturn4(), "Hello");
+    assertEqual(testReachabilityWithIfContainingReturn5(), "Ballerina");
+    assertEqual(testReachabilityWithIfContainingReturn6(), "Ballerina");
+    assertEqual(testReachabilityWithIfContainingReturn7(), "Ballerina");
+}
+
+function testReachabilityWithIfContainingReturn() returns string {
```

Comment:
Usual pattern would be to have something like `reachabilityWithIfContainingReturn` be the name of the function with this logic and then `testReachabilityWithIfContainingReturn` calling `reachabilityWithIfContainingReturn` and doing the assertion.

Reply:
IMO `reachabilityWithIfContainingReturn` does not imply action. Every function is an action, so the name should contain at least one verb right?

Reply:
What I meant to say was that the "test" prefix should be for the function in which the assertion is done.